Source/items.cpp

```diff
@@ -2375,8 +2371,21 @@ void CalcPlrItemVals(Player &player, bool loadgfx)
 				dmod += item._iPLDamMod;
 				ghit += item._iPLGetHit;
 				lrad += item._iPLLight;
-				ihp += item._iPLHP;
-				imana += item._iPLMana;
+
+				// Check for Acolyte's Amulet and Gladiator Ring to apply bonuses as life and mana changes, rather than getting static bonuses from the item data
+				if (IsAnyOf(item._iDamAcFlags, ItemSpecialEffectHf::ManaToLife)) {
+					int portion = ((player._pMaxManaBase >> 6) * 50 / 100) << 6;
+					imana -= portion;
+					ihp += portion;
+				} else if (IsAnyOf(item._iDamAcFlags, ItemSpecialEffectHf::LifeToMana)) {
+					int portion = ((player._pMaxHPBase >> 6) * 40 / 100) << 6;
+					ihp -= portion;
+					imana += portion;
+				} else {

```

Comment:
Should this not be included in the if/else? wouldn't you want to apply item bonuses after doing the ManaToLife or LifeToMana? Also, what if you are wearing both?

Reply:
Nevermind my bad, I was thinking this was the player loop, this is looping through items, so it shouldn't be an issue.

Reply:
This wouldn't be a problem under normal circumstances, but for the sake of modding, if someone wanted to put this effect on an item as well as another effect that modifies hp and/or life, this would be a problem.